packages/destination-actions/src/destinations/engage/utils/MessageSendPerformer.ts

```diff
@@ -0,0 +1,319 @@
+    return res
+  }
+
+  /**
+   * check if the externalId object is supported for sending a message by current class of message sender
+   * @param externalId
+   * @returns
+   */
+  abstract isSupportedExternalId(externalId: NonNullable<TPayload['externalIds']>[number]): boolean
+
+  static readonly nonSendableStatuses = ['unsubscribed', 'did not subscribed', 'false'] // do we need that??
+  static readonly sendableStatuses = ['subscribed', 'true']
+
+  /**
+   * allows access to static members of the current class that can be overriden in subclasses
+   */
+  getStaticMembersOfThisClass<
+    ThisStaticClass extends typeof MessageSendPerformer = typeof MessageSendPerformer
+  >(): ThisStaticClass {
+    return this.constructor as ThisStaticClass
+  }
+
+  /**
+   * check if extId is (un)subscribed (returns true|false) or if subscription status is invalid (returns undefined)
+   */
+  isExternalIdSubscribed(extId: ExtId<TPayload>): boolean | undefined {
+    const staticMems = this.getStaticMembersOfThisClass()
+    const subStatus = extId.subscriptionStatus?.toString()?.toLowerCase()
+    if (!subStatus) return false // falsy status is valid and considered to be Not Subscribed, so return false
+    // if subStatus is not in any of the lists of valid statuses, then return true
+    if (staticMems.sendableStatuses.includes(subStatus)) return true
+    if (staticMems.nonSendableStatuses.includes(subStatus)) return false
+    return undefined //Invalid subscriptionStatus
+  }
+
+  /**
+   * Gets all sendable recepients for the current payload or a reason why it is not sendable
+   * @returns
+   */
+  getSendabilityPayload(): SendabilityPayload<TPayload> {
+    if (!this.payload.send) {
+      return { sendabilityStatus: SendabilityStatus.SendDisabled }
+    }
+
+    // list of extenalIds that are supported by this Channel, if none - exit
+    const supportedExtIdsWithSub = this.payload.externalIds
+      ?.filter((extId) => this.isSupportedExternalId(extId))
+      .map((extId) => ({ extId, isSubscribed: this.isExternalIdSubscribed(extId) }))
+    if (!supportedExtIdsWithSub || !supportedExtIdsWithSub.length)
+      return {
+        sendabilityStatus: SendabilityStatus.NoSupportedExternalIds
+      }
+
+    const invalidSubStatuses = supportedExtIdsWithSub.filter((e) => e.isSubscribed === undefined).map((e) => e.extId)
+
+    const subscribedExtIds = supportedExtIdsWithSub.filter((e) => e.isSubscribed === true).map((e) => e.extId)
+
+    if (!subscribedExtIds.length) {
+      return invalidSubStatuses.length == 0
+        ? {
+            sendabilityStatus: SendabilityStatus.NotSubscribed
+          }
+        : {
+            sendabilityStatus: SendabilityStatus.InvalidSubscriptionStatus,
+            invalid: invalidSubStatuses
+          }
+    }
+
+    // if have subscribed, then return them IF they have id values (e.g. phone numbers)
+    if (subscribedExtIds.length) {
+      // making sure subscribed have phone numbers
```

Comment:
nit since this applies to more than just phone 
```suggestion
      // making sure subscribed have phone numbers, push tokens, etc.
```